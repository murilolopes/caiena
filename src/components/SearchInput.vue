<template>
  <div>
    <input type="text" v-model="term" @keyup.enter="search" />
  </div>
</template>

<script>
export default {
  name: "SearchInput",
  data() {
    return {
      term: "",
    };
  },
  methods: {
    search() {
      this.$store.dispatch("search", this.term, { root: true });
    },
  },
};
</script>

<style></style>
