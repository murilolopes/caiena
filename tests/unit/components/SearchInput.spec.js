import { mount } from "@vue/test-utils";
import SearchInput from "@/components/SearchInput.vue";

describe("SearchInput", () => {
  it("renders correctly", () => {
    const wrapper = mount(SearchInput);
    expect(wrapper.element).toMatchSnapshot();
  });
});
