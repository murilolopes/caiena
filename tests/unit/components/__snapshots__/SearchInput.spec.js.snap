// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchInput renders correctly 1`] = `
<div>
  <input
    type="text"
  />
</div>
`;
